shared/logger.go

```diff
@@ -143,14 +154,35 @@ func newAppEngineFlexContext(r *http.Request, project string, commonResource *mr
 // commonResource is an optional override to the monitored resource details appended to each log.
 // e.g. in the Flex environment, it pays to override this value to type gae_app, to ensure finding
 // logs is consistent between services.
-func HandleWithGoogleCloudLogging(h http.HandlerFunc, project string, commonResource *mrpb.MonitoredResource) http.HandlerFunc {
+func HandleWithGoogleCloudLogging(h http.HandlerFunc, project string, childLogger, parentLogger *gclog.Logger) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx, err := newAppEngineFlexContext(r, project, commonResource)
-		if err != nil {
-			h(w, r)
-			return
+		start := time.Now()
+		// See https://cloud.google.com/appengine/docs/flexible/go/writing-application-logs
+		traceID := strings.Split(r.Header.Get("X-Cloud-Trace-Context"), "/")[0]
+		if traceID != "" {
+			traceID = fmt.Sprintf("projects/%s/traces/%s", project, traceID)
 		}
-		h(w, r.WithContext(ctx))
+
+		gcl := gcLogger{
+			logger:      childLogger,
+			traceID:     traceID,
+			maxSeverity: gclog.Default,
+		}
+		h(gcResponseWriter{gcLogger: &gcl, w: w}, r.WithContext(withLogger(r.Context(), &gcl)))
+
+		end := time.Now()
+		e := gclog.Entry{
+			Timestamp: end,
+			Trace:     traceID,
+			Severity:  gcl.maxSeverity,
+			HTTPRequest: &gclog.HTTPRequest{
+				Request: r,
+				Status:  gcl.statusCode,
```

Comment:
@KyleJu FYI I've fixed this by wrapping the `ResponseWriter`.

Reply:
Thanks for the fix! The same fix would work for https://github.com/web-platform-tests/wpt.fyi/pull/2208. I was just thinking if there could be any problem when HttpResponder and context shared the same instance of a Logger but nothing came up. Very clean fix nevertheless!